.github/workflows/typescript.yml

```diff
@@ -0,0 +1,18 @@
+name: typescript
+on: push
+
+jobs:
+    typescript:
+        runs-on: ubuntu-latest
+        steps:
+            - uses: actions/checkout@v2
+            - uses: guardian/actions-setup-node@main
+
+            # Cache npm dependencies using https://github.com/bahmutov/npm-install
```

Comment:
This looks like a useful action!

Out of interest (and not worth stressing about here), do we as a department have guidelines about when we use marketplace actions and when we should stick to the built in ones (with a little more config)?

This is achievable with the GitHub cache action, which is:

## a lot more flexible and powerful

This means we'd be able to use the same tool for lots of workflows, rather than having to use different bespoke ones each time.

## supported by GitHub

This is useful in both operational and security terms. It'll certainly keep working, and it'll certainly be kept up to date with security patches. (It's also much more likely to have its source code stay secure)

## harder to use

Obviously for this workflow the bahmutov action is loads simpler, so that's a huge plus. I wonder if the actions folks want to set some guidelines that will help teams balance these concerns?

Reply:
Hey great question Adam! I don't think we do have any guidelines currently. For full disclosure I ~~stole~~ borrowed this from the DCR TypeScript action config as I was interested to see the impact it had (with the idea that if this was successful, we could migrate other actions over). I'd be up for switching over to the GitHub cache action as an experiment to see how that looks. Either before or after merging this.

Reply:
We chatted about this in the Client Side Infra meeting today. There aren't any established guidelines around actions and when to pick 3rd party or GitHub maintained. The action in question here is being used on other repos and there's a high level of trust around it. We talked about possibly maintaining a list of recommended actions for consistency across repos. One thing I learned is that we can [keep our actions up to date with Dependabot](https://docs.github.com/en/code-security/supply-chain-security/keeping-your-dependencies-updated-automatically/keeping-your-actions-up-to-date-with-dependabot) which is great. That brings the confidence we have in keeping our actions up to date in line with other library dependencies.